agent/consul/prepared_query_endpoint.go

```diff
@@ -393,6 +393,24 @@ func (p *PreparedQuery) Execute(args *structs.PreparedQueryExecuteRequest,
 	// Respect the magic "_agent" flag.
 	if qs.Node == "_agent" {
 		qs.Node = args.Agent.Node
+	} else if qs.Node == "_ip" {
+		if args.Source.Ip != "" {
+			_, nodes, err := state.Nodes(nil)
+			if err == nil {
+				for _, node := range nodes {
+					if args.Source.Ip == node.Address {
+						qs.Node = node.Node
+						break
+					}
+				}
+			}
+		} 
+		
+		// Either a source IP was given but we couldnt find the associated node
+		// or no source ip was given. In both cases we should wipe the Node value
+		if qs.Node == "_ip" {
```

Comment:
For the one case of _ip is set but there is no source IP I am going to make it return an error. I think for the other case we want to silently ignore it just like we do if we dont have coordinates for the node name provided. Otherwise if someone creates a prepared query any time they execute that query from a host that doesn't have a node registered for it they are going to get a log message.

Reply:
One other note: Do we currently support having a cluster with mixed versions. Because we could get into a scenario where a server is at 1.0.7 but there is some agent still sitting at 1.0.6. That 1.0.6 agent wont be sending the source IP to the server and it would cause the error. We may still want to error out. I am not sure. Just pointing out that if a user executes a prepared query originating from a <1.0.7 node to a 1.0.7 server then it will error out every time instead of executing the query the same as it would have for the previous version.

Reply:
Ok, I can add the warning message then.

Reply:
As for the second scenario where we have the source IP but it doesn't map to any node, I dont see a reason for it to log anything. It currently will silently ignore it just like it would if we didn't have coordinates for a particular node or the dcs of the agent and server were not equal.